### PR TITLE
Use batched operations in ps

### PR DESCRIPTION
**Pull request by baude**
Because the podman ps command has to collection a lot of information
from various places, many of which are controlled by locks, it is a
good candidate for doing batch operations under a single lock.

Signed-off-by: baude <bbaude@redhat.com>

### CODE REVIEW

File: cmd/podman/ps.go [hunk 1]
```diff
@@ -374,22 +374,34 @@ func (p *psTemplateParams) headerMap() map[string]string {
 // getTemplateOutput returns the modified container information
 func getTemplateOutput(containers []*libpod.Container, opts psOptions) ([]psTemplateParams, error) {
 	var psOutput []psTemplateParams
-	var status string
+	var status, ctrID string
```
Comment (rhatdan) [inline cmd/podman/ps.go]: nit: Can you put these into one big var block
```
var (
psOutput []psTemplateParams
status, ctrID string
...
)
```

**Comment by TomSweeneyRedHat** [comment]
@baude gofmt isn't happy with you, otherwise LGTM

**Comment by mheon** [comment]
I'd like to see the var grouping @rhatdan addressed but otherwise LGTM

File: cmd/podman/ps.go (hunk 1 above)
Comment (baude) [reply cmd/podman/ps.go]: fixed

**Comment by mheon** [comment]
@rh-atomic-bot r+

**Comment by mheon** [comment]
@rh-atomic-bot retry

**Comment by baude** [comment]
@rh-atomic-bot retry

**Comment by mheon** [comment]
@rh-atomic-bot retry

**Comment by rhatdan** [comment]
@rh-atomic-bot r+

**Comment by rhatdan** [comment]
@rh-atomic-bot retry

**Comment by mheon** [comment]
@rhatdan I think homu is having issues with its Fedora repos.

**Comment by baude** [comment]
@rh-atomic-bot retry

**Comment by rhatdan** [comment]
Bot seems to be hung.  Merging manually.

**Pull request merged**